routing/index_graph_starter.cpp

```diff
@@ -4,186 +4,96 @@
 
 namespace routing
 {
-IndexGraphStarter::IndexGraphStarter(IndexGraph & graph, RoadPoint const & startPoint,
-                                     RoadPoint const & finishPoint)
-  : m_graph(graph)
-  , m_start(graph, startPoint, graph.GetNumJoints())
-  , m_finish(graph, finishPoint, graph.GetNumJoints() + 1)
+IndexGraphStarter::IndexGraphStarter(IndexGraph & graph, FakeVertex const & start,
+                                     FakeVertex const & finish)
+  : m_graph(graph), m_start(start, false /* isFinish */), m_finish(finish, true /* isFinish */)
```

Comment:
The comment "isFinish" is wrong here, as this parameter means "forward" or "backward" in the context of segment id.